### ISSUE TITLE
Bug - Text - Header does not read as "heading" for screen reader

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/vets-design-system-documentation#5364]
## Configuring this issue
- [ ] Add issue to appropriate epic
## Description
As part of ticket #5345 a bug was noted with the Text component caused by [PR 631](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/631) where they deleted the intentional functionality of header text applying an accessibility role of "heading". The associated bug was [ticket 630](https://github.com/department-of-veterans-affairs/va-mobile-library/issues/630), but testing locally with reverting the changes of the aforementioned PR this was not bugged at all and likely another bug caused by not using a physical iOS device to test where the simulator on Mac is not a faithful representation. For completeness, they also made some related changes on [PR 621](https://github.com/department-of-veterans-affairs/va-mobile-library/pull/621) which is believed to, again, not have been bugged on any actual iOS device.

## Tasks
- [ ] Fix Text component for screen reader to correctly read header text as a heading
- [ ] Update Alert component to not override the Text component's a11y for functionally identical behavior

## Acceptance Criteria
- [ ] Text component correctly reads headers as a heading in the screen reader
- [ ] Alert component retains reading headers for non-expandable Alerts as headings without duplicate setting of the a11y role
